### PR TITLE
Fix EKS ci

### PR DESCRIPTION
**Pull request by gurevichdmitry**
### Summary of your changes

This PR enables the post-merge event and also addresses the issue where the test pod wasn't reflecting test results in the workflow run.

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

- [ ] Generate rule metadata using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rules-metadata)
- [ ] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

### CODE REVIEW

File: tests/test_environments/k8s-cloudbeat-tests/templates/tests/cloudbeat-test-pod.yml [hunk 1]
```diff
@@ -16,11 +16,13 @@ spec:
     - /bin/sh
     - -c
     - |
-      pytest -rA --disable-warnings ${MARKER_CMD} "${TEST_MARKER}" ${RANGE_FLAG} ${TEST_RANGE} --alluredir=/usr/src/app/report
+      pytest -rA --disable-warnings ${MARKER_CMD} "${TEST_MARKER}" ${RANGE_FLAG} ${TEST_RANGE} --alluredir=/usr/src/app/report && pytest_exit_status=$? || pytest_exit_status=$?
```
Comment (orestisfl) [inline tests/test_environments/k8s-cloudbeat-tests/templates/tests/cloudbeat-test-pod.yml:19]:
```suggestion
      pytest -rA --disable-warnings ${MARKER_CMD} "${TEST_MARKER}" ${RANGE_FLAG} ${TEST_RANGE} --alluredir=/usr/src/app/report; pytest_exit_status=$?
```

File: tests/test_environments/k8s-cloudbeat-tests/templates/tests/cloudbeat-test-pod.yml [hunk 2]
```diff
@@ -16,11 +16,13 @@ spec:
...
       # When directly linking to mounted path,
       # INTERNALERROR> OSError: [Errno 16] Device or resource busy: '/usr/src/app/tests/report'
       # To workaround this, copy the report to a different path
-      cp -rv /usr/src/app/report/* /usr/src/app/tests/report
+      cp -r /usr/src/app/report/* /usr/src/app/tests/report
```
Comment (orestisfl) [inline tests/test_environments/k8s-cloudbeat-tests/templates/tests/cloudbeat-test-pod.yml:23]:
```suggestion
      cp -r /usr/src/app/report/* /usr/src/app/tests/report || exit $?
```

for this command's failure to be accounted for as well

File: .github/workflows/eks-ci.yml [hunk 3]
```diff
@@ -9,6 +9,13 @@ on:
         description: "Specify test markers to run"
         default: '["eks"]'
 
+  # Added to check EKS tests run on PR
+  pull_request:
+    branches:
+      - main
+      - "[0-9]+.[0-9]+"
+    types: [ opened, synchronize, reopened ]
+
```
Comment (orestisfl) [inline .github/workflows/eks-ci.yml:18]:
```suggestion
```

Was this intended to be temporary?

File: tests/test_environments/k8s-cloudbeat-tests/templates/cloudbeat-eks-pytest.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.cloudbeatEks.test_environments -}}
+{{- if .Values.cloudbeatEks.deploy -}}
```
Comment (orestisfl) [inline tests/test_environments/k8s-cloudbeat-tests/templates/cloudbeat-eks-pytest.yaml:1]: Is this intended?

Comment (gurevichdmitry) [reply tests/test_environments/k8s-cloudbeat-tests/templates/cloudbeat-eks-pytest.yaml:1]: Yes, this fix is the correct one, taken from the Helm values.

**Comment by gurevichdmitry** [comment]
EKS test runs [evidence](https://github.com/elastic/cloudbeat/actions/runs/7988291954/job/21812787818)

**Review by orestisfl — APPROVED** [review]

**Pull request merged**